src/Controls/samples/Controls.Sample/Pages/Controls/MapsGalleries/MapPinsGallery.xaml.cs

```diff
@@ -100,13 +106,34 @@ void OnAdd10PinsClicked(object sender, EventArgs e)
 
 		void AddPin()
 		{
-			pinsMap.Pins.Add(new Pin()
+			var randomLocation = GetRandomLocation();
+			var pin = new Pin
 			{
 				Label = $"Location {_locationIncrement++}",
-				Location = _randomLocations[_locationRandomSeed.Next(0, _randomLocations.Length)],
-			});
+				Location = randomLocation,
+			};
+			pinsMap.Pins.Add(pin);
+			MoveMapTo(randomLocation);
+		}
+
+		void MovePin()
+		{
+			if (pinsMap.Pins.Count == 0)
+			{
+				return;
+			}
+
+			var randomLocation = GetRandomLocation();
+			pinsMap.Pins[0].Location = randomLocation;
+			MoveMapTo(randomLocation);
 		}
 
+		Position GetRandomLocation() =>
+			_randomLocations[_locationRandomSeed.Next(_randomLocations.Length)];
+
+		void MoveMapTo(Position location) =>
+			pinsMap.MoveToRegion(MapSpan.FromCenterAndRadius(location, Distance.FromKilometers(5)));
```

Comment:
Could be nice to create a constant:
`const double DefaultMapRadiusKm = 5.0;`

Reply:
@jsuarezruiz . Done . Please let me know if you have concerns.